client/consul.go

```diff
@@ -34,19 +34,39 @@ func NewConsulAPI() (*APIClient, error) {
 		headers["X-Consul-Token"] = token
 	}
 
+	httpClient, err := NewConsulHTTPClient()
+	if err != nil {
+		return nil, err
+	}
+
+	apiClient := NewAPIClientFromHTTP("consul", addr, headers, httpClient)
+
+	return apiClient, nil
+}
+
+func NewConsulHTTPClient() (*http.Client, error) {
 	tlsConfig, err := NewConsulTLSConfig()
 	if err != nil {
 		return nil, err
 	}
 
-	apiClient := NewAPIClient("consul", addr, headers)
+	httpClient, err := NewHTTPClientWithTLS(tlsConfig)
+	if err != nil {
+		return nil, err
+	}
 
-	err = configureHttpClientTLS(apiClient.http.(*http.Client), tlsConfig)
+	return httpClient, nil
+}
+
+func NewHTTPClientWithTLS(t *TLSConfig) (*http.Client, error) {
```

Comment:
This isn't the file where this function would ultimately live, but I stuck it here to start with and forgot to move it before pushing up a work in progress commit.